pkg/odo/util/completion/completionhandlers_test.go

```diff
@@ -18,6 +20,201 @@ import (
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan:  []scv1beta1.ClusterServicePlan{testingutil.FakeClusterServicePlan("default", 1)},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+				flagValues: map[string]string{"plan": "other"},
+			},
+			output: []string{},
+		},
+		{
+			name:                 "Case 4: matching plan supplied and no other parameters supplied",
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan: []scv1beta1.ClusterServicePlan{
+				testingutil.FakeClusterServicePlan("plan2", 2),
+				testingutil.FakeClusterServicePlan("plan1", 1),
+			},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+				flagValues: map[string]string{"plan": "plan1"},
+			},
+			output: []string{"PLAN_DATABASE_URI", "PLAN_DATABASE_USERNAME", "PLAN_DATABASE_PASSWORD", "SOME_OTHER"},
+		},
+		{
+			name:                 "Case 5: no plan supplied but some other parameters supplied",
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan:  []scv1beta1.ClusterServicePlan{testingutil.FakeClusterServicePlan("default", 1)},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+				flagValues: map[string]string{"parameters": "[PLAN_DATABASE_USERNAME, SOME_OTHER]"},
+			},
+			output: []string{"PLAN_DATABASE_URI", "PLAN_DATABASE_PASSWORD"},
+		},
+		{
+			name:                 "Case 6: matching plan supplied but some other parameters supplied",
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan:  []scv1beta1.ClusterServicePlan{testingutil.FakeClusterServicePlan("default", 1)},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+				flagValues: map[string]string{"plan": "default", "parameters": "[PLAN_DATABASE_USERNAME]"},
+			},
+			output: []string{"PLAN_DATABASE_URI", "PLAN_DATABASE_PASSWORD", "SOME_OTHER"},
+		},
+	}
+
+	for _, tt := range tests {
+		client, fakeClientSet := occlient.FakeNew()
+		context := genericclioptions.NewFakeContext("project", "app", "component", client)
+
+		fakeClientSet.ServiceCatalogClientSet.PrependReactor("list", "clusterserviceclasses", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, tt.returnedServiceClass, nil
+		})
+
+		fakeClientSet.ServiceCatalogClientSet.PrependReactor("list", "clusterserviceplans", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, &scv1beta1.ClusterServicePlanList{Items: tt.returnedServicePlan}, nil
+		})
+
+		completions := ServiceParameterCompletionHandler(nil, tt.parsedArgs, context)
+		sort.Strings(completions)
+
+		// Sort the output and expected o/p in-order to avoid issues due to order as its not important
+		sort.Strings(completions)
```

Comment:
same here as above, why are we sorting twice?

Reply:
Fixed

Reply:
:+1: